how-to/add_controller.rst

```diff
@@ -1,119 +1,120 @@
 
-3. Next the controller should be put into HA mode (optional): 
+.. note::
 
-    ``juju enable-ha``
+    A Juju server's default certificate contains a `SAN <https://en.wikipedia.org/wiki/Subject_Alternative_Name>`__ for the name ``juju-apiserver``.
 
-4. The we switch to the controller model: 
+The use of the ``--local`` flag avoids the need to provide a public DNS address and ``--tls-hostname`` provides the expected
+hostname used in TLS, a useful way of handling TLS issues during local development. These config options are normally not needed
+in a production environment.
 
-    ``juju switch controller``
 
-5. Download the controller bundle from `here <https://drive.google.com/file/d/17GHATHXGg2GuIeIWGr0FvkguMRdv5vnH/view?usp=sharing>`__.
+LXD Controller
+--------------
 
-6. Extract the file: 
+The following section provides guidance on how to connect a controller bootstrapped on LXD to your JIMM running in MicroK8s.
 
-    ``tar xvf controller.tar.xz``
+The steps will be similar to those for adding a MicroK8s hosted controller but because we are traversing from the isolated network
+of the container through to LXD's network, there will be additional steps.
 
-7. Move to the controller folder: 
+Run the following commands to bootstrap a LXD based controller:
 
-    ``cd controller``
+.. code:: bash
 
-8. Deploy the bundle: 
+    CLOUDINIT_FILE="cloudinit-tweak.temp.yaml"
+    CONTROLLER_NAME="workload-lxd"
+    CLOUDINIT_TEMPLATE=$'cloudinit-userdata: |
+    preruncmd:
+        - echo "%s    test-jimm.localhost" >> /etc/hosts
+    ca-certs:
+        trusted:
+        - |\n%s'
+    printf "$CLOUDINIT_TEMPLATE" "$(lxc network get lxdbr0 ipv4.address | cut -f1 -d/)" "$(cat /usr/local/share/ca-certificates/jimm-test.crt | sed -e 's/^/\t  /')" > "${CLOUDINIT_FILE}"
+    juju bootstrap lxd "${CONTROLLER_NAME}" --config "${CLOUDINIT_FILE}" --config login-token-refresh-url=https://test-jimm.localhost/.well-known/jwks.json --debug 
 
-    ``juju deploy  ./bundle.yaml --overlay ./overlay-certbot.yaml --map-machines=existing``
+The set of commands will do the following:
 
-9. Once the bundle has been deployed, get the public IP of the ``haproxy/0`` unit: 
+- Create a cloud-init template, cloud-init provisions the LXD container that Juju will use.
+- The cloud-init script will create an entry in ``/etc/hosts`` to point ``test-jimm.localhost`` to the LXD bridge address in order to route this request to your host network.
+- The cloud-init script will add the CA cert in ``/usr/local/share/ca-certificates/jimm-test.crt`` to the machine. If you've placed JIMM's CA cert elsewhere, please update this file location.
+- Finally the bash script will bootstrap Juju and configure it to communicate with JIMM.
 
-    ``juju status  --format json | jq '.applications.haproxy.units["haproxy/0"]["public-address"]'``
+Next, we will create a network relay to forward traffic from our host network through to the Juju server running in a LXC container.
```

Comment:
s/LXC/LXD (let's just use one everywhere; LXD or LXC)

Reply:
I'd like to stick with LXD in most places because that's the term Juju uses. But it's not quite correct to say LXD container vs LXC Container. So I think it's okay to use LXC where appropriate.